test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -1046,6 +1051,145 @@ private void testSnapshotSyncAndLogEntrySync(int numCyclesToDelayApply, boolean
         verifyPersistedLogEntryMetadata();
     }
 
+     /**
+     * Test Log Entry (delta) Sync for the case where the ACKs are arbitrarily dropped
+     * for a fixed number of times at the Source. This will test that LR is
+     * (i) not impacted by dropped ACKs
+     * (ii) Source resends msgs for which it hasn't received the ACKs
+     * (iii) Sink handles the already seen and processed msgs.
+     **/
+    @Test
+    public void testLogEntrySyncWithAckDrops() throws Exception {
+        setupEnv();
+        // Write data in transaction to t0 and t1
+        Set<String> crossTables = new HashSet<>();
+        crossTables.add(t0);
+        crossTables.add(t1);
+
+        // Writes transactions to t0, t1 and t2 + transactions across 'crossTables'
+        writeCrossTableTransactions(crossTables, true);
+
+        // Start Log Entry Sync
+        expectedAckMessages = NUM_KEYS * WRITE_CYCLES;
+        testConfig.clear().setDropAckLevel(1);
+
+        startLogEntrySync(crossTables, WAIT.ON_ACK);
+
+        // Verify Data on Destination site
+        log.debug("****** Verify Data on Destination");
+        // Because t2 is not specified as a replicated table, we should not see it on the destination
+        srcDataForVerification.get(t2).clear();
+
+        // Verify Destination
+        verifyData(dstCorfuTables, srcDataForVerification);
+    }
+
+
+     /**
+     * Test Log Entry (delta) Sync for the case where messages are arbitrarily dropped at the
+     * destination and ACKs are arbitrarily dropped at the source. This tests
+     * (i) messages dropped at the destination is resent. This would also test that Sink handles out of order messages.
+     * (ii) messages for which ACKs were dropped at the source are resent and Sink handles already processed data.
+     */
+    @Test
+    public void testLogEntrySyncWithMsgDropsAndAckDrops() throws Exception {
```

Comment:
I tried running the tests on current master and all new tests pass except testLogEntrySyncWithFSMChangeAndWithAckDrop, I was expecting that testLogEntrySyncWithAckDrops would fail. For the test to be correct, shouldn't we be writing additional data after the ACK drop so we are sure that it continues to progress? cause as the test is right now its only validating the data on standby and the ack was lost but the data was persisted.

Reply:
I can include additional data write as well, but even with the current test, there are logs that are read and replicated after ACK drops. I can maybe have an assert in between to showcase that

Reply:
Ok, so that means the amount of data we write generates more than one batch. But still, shouldn't the test fail on current master?

Reply:
No, the test should fail on master because we currently have a mechanism where active keeps resending a msg to standby until it receives an ACK for the msg or receives an ACK for a msg with higher ts which would imply that standby has processed the msg. 

Issue is only when the active looses the information that it has sent a msg to standby and resends the same msg but now that msg includes new entries...like the scenario in the desc.

Reply:
testLogEntrySyncWithFSMChangeAndWithAckDrop() should fail on master